modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -1471,7 +1498,30 @@ static const iocshFuncDef iocshLoadFuncDef = {"iocshLoad",2,iocshLoadArgs,
                                               "  * (optional) replace macros within the file with provided values\n"};
 static void iocshLoadCallFunc(const iocshArgBuf *args)
 {
-    iocshSetError(iocshLoad(args[0].sval, args[1].sval));
+    iocshLoad(args[0].sval, args[1].sval);
+}
+
+/* set*/
+static const iocshArg iocshSetArg0 = { "pathname",iocshArgString};
+static const iocshArg iocshSetArg1 = { "macros", iocshArgString};
+static const iocshArg *iocshSetArgs[2] = {&iocshSetArg0, &iocshSetArg1};
+static const iocshFuncDef iocshSetFuncDef = {"set",2,iocshSetArgs,
+                                              "Takes local macro name and value\n"
+                                              "Sets local macro with provided value\n"};
+static void iocshSetCallFunc(const iocshArgBuf *args)
+{
+    iocshSet(args[0].sval, args[1].sval);
+}
+
+/* show */
+static const iocshArg iocshShowArg0 = { "pathname",iocshArgString};
+static const iocshArg iocshShowArg1 = { "macros", iocshArgString};
+static const iocshArg *iocshShowArgs[2] = {&iocshShowArg0, &iocshShowArg1};
+static const iocshFuncDef iocshShowFuncDef = {"show",2,iocshShowArgs,
+                                              "Shows all set local marcros\n"};
+static void iocshShowCallFunc(const iocshArgBuf *args)
+{
+    iocshShow();
```

Comment:
The metadata of these two commands is incomplete. The `show` command takes no arguments, and `iocshShowArgs` should reflect that. The `set` command *does* take two arguments, but the first one is no path, and the second one is not a list of macros. Please update this documentation.

I would avoid refering to "macros" for these user-facing descriptions. From the user's point of view, they are dealing with iocsh variables and don't care that the underlying implementation relies on the macro machinery.